main.cpp

```diff
@@ -173,24 +240,63 @@ int main (int argc, char** argv)
 			current_handle = current_handle->next;
 		}
 
-
-		if (needrefresh)
-		{
-			if ((!DEBUG)&&(!tracemode))
+		time_t const now = ::time(NULL);
+ 		if( last_refresh_time + refreshdelay <= now )
+ 		{
+ 			last_refresh_time = now;
+ 			if ((!DEBUG)&&(!tracemode))
 			{
 				// handle user input
 				ui_tick();
 			}
 			do_refresh();
-			needrefresh = false;
 		}
-
-		// If no packets were read at all this iteration, pause to prevent 100%
-		// CPU utilisation;
+ 
+		//if not packets, do a select() until next packet
```

Comment:
not sure, if it may block waiting for new packets ignoring the ones already waiting